src/main/java/core/basesyntax/service/FruitServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FruitServiceImpl implements FruitService {
+    private static final int REMOVE_NAMES_INDEX = 0;
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<String[]> report(String data) {
```

Comment:
this method and service has too many responsibilities:
* parses string into a transaction
* processes the data inside it
* saves it to somewhere
* builds a report

All these points should be separate services, like `TransactionParser`, `DataProcessor`(uses strategy, because I do not see one in your solution at all), `ReportProvider`, and called in main in respective order.
In general try to "dive deep" in the common mistakes and tips in the task description, especially in the diagram there. Ask any questions that you may have.
Do that first and then we will proceed with a more detailed review :wink: